micro_sam/evaluation/inference.py

```diff
@@ -392,118 +389,127 @@ def _save_segmentation(masks, prediction_path):
     shape = masks.shape[-2:]
     masks = [{"segmentation": mask, "area": mask.sum()} for mask in masks]
     segmentation = mask_data_to_segmentation(masks, shape, with_background=True)
-    imageio.imwrite(prediction_path, segmentation)
+    imageio.imwrite(prediction_path, segmentation, compression=5)
+
+
+def extract_instances_from_batched_outputs(batched_outputs):
+    masks = [m["segmentation"][None] for m in batched_outputs]
+    masks = torch.stack(masks)
+    masks = (masks > 0.5).to(torch.float32)
```

Comment:
The masks are already binary, so you don't need this. Overall, this is simple enough to write it as a one-liner and then you don't need to refactor this into a function.

Reply:
See comment below.

Reply:
Ahha yes, that's correct. Fixed it, thanks!